<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="../../img/favicon.ico">
        <title>Basics - Bioinformatics</title>
        <link href="../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../css/font-awesome.min.css" rel="stylesheet">
        <link href="../../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/color-brewer.min.css">
        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
        <![endif]-->

        <script src="../../js/jquery-1.10.2.min.js" defer></script>
        <script src="../../js/bootstrap-3.0.3.min.js" defer></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script> 
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
            <div class="container">

                <!-- Collapsed navigation -->
                <div class="navbar-header">
                    <!-- Expander button -->
                    <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                        <span class="sr-only">Toggle navigation</span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                    </button>
                    <a class="navbar-brand" href="../..">Bioinformatics</a>
                </div>

                <!-- Expanded navigation -->
                <div class="navbar-collapse collapse">
                        <!-- Main navigation -->
                        <ul class="nav navbar-nav">
                            <li class="dropdown">
                                <a href="#" class="dropdown-toggle" data-toggle="dropdown">Shell <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
  <li class="dropdown-submenu">
    <a href="#">Basics/Hacking</a>
    <ul class="dropdown-menu">
            
<li >
    <a href="../../shell/shell/">Commands</a>
</li>
            
<li >
    <a href="../../shell/shell_adv/">Cloud Computing</a>
</li>
            
<li >
    <a href="../../shell/ubuntu_server/">Ubuntu Server</a>
</li>
            
<li >
    <a href="../../shell/gpu/">GPUs</a>
</li>
    </ul>
  </li>
                                    
<li >
    <a href="../../shell/fasta/">File Manipulation (fasta)</a>
</li>
                                    
<li >
    <a href="../../shell/git/">Git</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="dropdown-toggle" data-toggle="dropdown">Python <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li >
    <a href="../../python/python/">Basics</a>
</li>
                                    
<li >
    <a href="../../python/bioconda/">Bioconda</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown active">
                                <a href="#" class="dropdown-toggle" data-toggle="dropdown">R <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li class="active">
    <a href="./">Basics</a>
</li>
                                    
  <li class="dropdown-submenu">
    <a href="#">Graphics</a>
    <ul class="dropdown-menu">
            
<li >
    <a href="../ggplot/">ggplot</a>
</li>
            
<li >
    <a href="../distanceheatmap/">heatmap</a>
</li>
    </ul>
  </li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="dropdown-toggle" data-toggle="dropdown">Tools <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li >
    <a href="../../tools/atom/">Atom Editor</a>
</li>
                                    
<li >
    <a href="../../tools/emboss/">EMBOSS</a>
</li>
                                    
<li >
    <a href="../../tools/mkdocs/">mkdocs</a>
</li>
                                    
<li >
    <a href="../../tools/UGENE/">UGENE</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="dropdown-toggle" data-toggle="dropdown">Sequencing <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li >
    <a href="../../sequencing/programs/">Programs</a>
</li>
                                    
<li >
    <a href="../../sequencing/illumina/">HTS (Illumina)</a>
</li>
                                    
<li >
    <a href="../../sequencing/nanopore/">Longreads (Nanopore)</a>
</li>
                                    
<li >
    <a href="../../sequencing/hybrid/">Hybridassembly</a>
</li>
                                    
<li >
    <a href="../../sequencing/annotation/">Annotation</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="dropdown-toggle" data-toggle="dropdown">Metagenome <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li >
    <a href="../../metagenome/metagenomics/">Metagenomics</a>
</li>
                                    
<li >
    <a href="../../metagenome/metabarcoding/">Metabarcoding</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="dropdown-toggle" data-toggle="dropdown">Docker <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li >
    <a href="../../docker/docker/">Docker-Basics</a>
</li>
                                    
<li >
    <a href="../../docker/win_docker/">Docker in Win</a>
</li>
                                    
<li >
    <a href="../../docker/gpu/">GPU_Docker</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="dropdown-toggle" data-toggle="dropdown">Other Topics <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li >
    <a href="../../other_topics/databases/">Databases</a>
</li>
                                    
<li >
    <a href="../../other_topics/gwas/">GWAS</a>
</li>
                                    
<li >
    <a href="../../other_topics/gs/">GS</a>
</li>
                                    
<li >
    <a href="../../other_topics/RNA_seq/">RNA seq</a>
</li>
                                    
  <li class="dropdown-submenu">
    <a href="#">Congress</a>
    <ul class="dropdown-menu">
            
<li >
    <a href="../../congress/LR_2019/">2019_LRM</a>
</li>
    </ul>
  </li>
                                </ul>
                            </li>
                        </ul>

                    <ul class="nav navbar-nav navbar-right">
                        <li>
                            <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                                <i class="fa fa-search"></i> Search
                            </a>
                        </li>
                            <li >
                                <a rel="next" href="../../python/bioconda/">
                                    <i class="fa fa-arrow-left"></i> Previous
                                </a>
                            </li>
                            <li >
                                <a rel="prev" href="../ggplot/">
                                    Next <i class="fa fa-arrow-right"></i>
                                </a>
                            </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
                <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
        <li class="main active"><a href="#overview-of-r">Overview of R</a></li>
            <li><a href="#installing-r-base-language-and-r-studio-environment">Installing r-base (language) and R-Studio (environment)</a></li>
            <li><a href="#basics">Basics</a></li>
            <li><a href="#data-storage-vector-matrix-data-frame-list">Data storage (Vector, Matrix, Data frame, List)</a></li>
        <li class="main "><a href="#advanced-data-manipulation">Advanced data manipulation</a></li>
            <li><a href="#if-else-for-while-function">if -else, for, while function</a></li>
            <li><a href="#working-dir">Working dir</a></li>
            <li><a href="#file-import-and-download">File import and download</a></li>
            <li><a href="#merging-two-files">Merging two files</a></li>
        <li class="main "><a href="#graphs-in-r">Graphs in R</a></li>
    </ul>
</div></div>
                <div class="col-md-9" role="main">

<h1 id="overview-of-r">Overview of R</h1>
<h2 id="installing-r-base-language-and-r-studio-environment">Installing r-base (language) and R-Studio (environment)</h2>
<ul>
<li>for Linux Mint (Xenial)</li>
</ul>
<pre><code class="bash"># Installing R language
sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys E298A3A825C0D65DFD57CBB651716619E084DAB9
sudo add-apt-repository 'deb [arch=amd64,i386] https://cran.rstudio.com/bin/linux/ubuntu xenial/'
sudo apt-get update
sudo apt-get install r-base
</code></pre>

<p>Get the R-Studio from <a href="https://www.rstudio.com/products/rstudio/download/#download">here</a>.</p>
<p>If you want to code in terminal type <code>R</code>. Now you are in a R environment ("console"). <code>q()</code> to exit R in terminal. <strong>But use R Studio for scripting not the terminal.</strong></p>
<p><strong>Installing packages in R</strong></p>
<p>Packages are organized in repositories:
<em> <a href="https://cran.r-project.org">CRAN</a>,
</em> <a href="https://www.bioconductor.org/">Bioconductor</a>,
<em> <a href="http://r-forge.r-project.org/">R-forge</a>,
</em> <a href="https://github.com">Github</a> or
* <a href="https://code.google.com">Googlecode</a></p>
<p>Installing a package in R and first time source location</p>
<pre><code class="R"># 1 establish bioconductor as a source, first time only
source(&quot;https://bioconductor.org/biocLite.R&quot;)

# 2 install package methylKit from biocLite
biocLite(&quot;methylKit&quot;)

# CRAN
# install from CRAN
install.packages(&quot;fortunes&quot;)
</code></pre>

<h2 id="basics">Basics</h2>
<p>"file" -&gt; "new project" -&gt; "new directory" and save it to a location</p>
<p>"file" -&gt; "new file" -&gt; "R Script"</p>
<ul>
<li>The output of R is given as <code>[1]</code> in the console</li>
</ul>
<p><strong>Help functions</strong>
Get help in R <a href="http://www.rseek.org/">here</a> or you use help commands:</p>
<pre><code class="R">help(read.table) # help on a function
?read.table # is the same

help.search(&quot;deviation&quot;)
??deviation # is the same
</code></pre>

<h3 id="variables">Variables</h3>
<ul>
<li>Variables: any letter or word incl. numbers, <code>_</code> or <code>.</code></li>
<li>Variables are case sensitive and don't have to be declared first</li>
</ul>
<pre><code class="R">x &lt;- 4+3 # adds sum of 4+3 to the variable x
y &lt;- 9 # also shown in the environment window or with ls()
x+y
[1] 16 # result of x+y
</code></pre>

<h3 id="operators">Operators</h3>
<p>More <a href="https://www.tutorialspoint.com/r/r_operators.htm">tutorials</a> about operators</p>
<pre><code class="R"># Arithmetic Operators
    +, -, *, /, %%, %/%, ^
# Relational Operators, TRUE or FALSE as a result
    &gt;, &lt;, &gt;=, &lt;=, ==, !=
# Logical Operators, &amp; = AND, | = OR
    &amp;, | , !, &amp;&amp;, ||      #&amp;&amp; = searches if the value is somewhere in the vectors
    4&lt;3 | 4&gt;3             #4&lt;3 = FALSE, 4&gt;3 = TRUE
    [1] TRUE
# Assignment Operators
    &lt;-, =, &lt;&lt;-, -&gt;, -&gt;&gt;
# Miscellaneous Operators
    %in%, %*%
# check help for more
</code></pre>

<h3 id="valuedata-types">Value/data types</h3>
<pre><code class="R"># Numeric – A floating point number (default data type of R):
    2.4, 9, 888
# Integer – A number with L suffix:
    2L, 25L, 0L
# Character - Any amount of text:
    &quot;a&quot;, &quot;school is a house&quot;, &quot;66&quot;
# Logical values:
    TRUE, FALSE
# Complex – defined via imaginary value i:
    2+3i, 5+8i
</code></pre>

<h3 id="class-and-typeof-functions">class() and typeof() functions</h3>
<pre><code class="R"># class( ) to determine the class of the R object
# typeof( ) to determine the internal type (storage mode) of the object
    typeof(5.67)
    [1] &quot;double&quot;
    class(5)
    [1] &quot;numeric&quot;
# To create integer variable in R, you invoke as.integer() function.
    x &lt;- as.integer(3)
    x
    [1] 3
    class(var_name)
    [1] &quot;integer&quot;
</code></pre>

<h3 id="print-and-paste-functions">print() and paste() functions</h3>
<pre><code class="R">print(&quot;I am at home&quot;) #prints its argument in R console
[1] &quot;I am at home&quot;
x &lt;- 6
print(x) #prints also variables
[1] 6
#paste() combines arguments and variables
print(paste(&quot;I went there&quot;, x, &quot;times&quot;))
[1] &quot;I went there 6 times&quot;
</code></pre>

<h2 id="data-storage-vector-matrix-data-frame-list">Data storage (Vector, Matrix, Data frame, List)</h2>
<ul>
<li><strong>Vectors</strong> are one row data storages<ul>
<li>all of the same class (e.g. <em>numeric</em>, <em>characters</em> like words or <em>logicals</em> like TRUE, FALSE)</li>
<li>allows arithmantics</li>
<li>access with <code>x[1]</code> or <code>x[c(1,2)]</code></li>
<li>remove data with <code>x[-1]</code> or <code>x[c(-1,-2)]</code></li>
</ul>
</li>
<li><strong>Matrices</strong> data table<ul>
<li>A table with values, all of the same class</li>
<li>access with <code>x[1,2]</code> <code>[,2]</code> or <code>x[1,c(1,2)]</code></li>
</ul>
</li>
<li><strong>Dataframes</strong> data table<ul>
<li>like a matrix but can store serveral classes into it</li>
<li>access with <code>x[1,2]</code> <code>[,2]</code> or <code>x[1,c(1,2)]</code></li>
</ul>
</li>
<li><strong>Lists</strong> object list<ul>
<li>can combine objects like vectors, matrices, dataframes and more in a single variable</li>
<li>each objects gets an identifier <code>[[1]]</code>, <code>[[2]]</code> and so on</li>
<li>access with <code>x[[2]][1]</code>: the first <code>[[ ]]</code> is for the list ID, the second <code>[ ]</code> is for the data in the list, its <code>[ ]</code> or <code>[,]</code> depending of stored vector or dataframe/matrix (see above)</li>
</ul>
</li>
</ul>
<hr />
<h3 id="vectors">Vectors</h3>
<ul>
<li>values are included with combine <code>c(&lt;value1&gt;, &lt;value2&gt;, &lt;etc.&gt;)</code></li>
</ul>
<pre><code class="R">x &lt;- c(2,3,5) #Vector containing three numeric values
x
[1] 2 3 5
class(x)
[1] &quot;numeric&quot; #numeric vector storage
</code></pre>

<p><strong>Vector commands</strong></p>
<p><code>length()</code> function gives length of vector
<code>class()</code> function gives value type
<code>typeof()</code> also does this somehow</p>
<p><strong>Combining vectors</strong>
* again with the <code>c()</code> command.</p>
<pre><code class="R">x &lt;- c(1,2,3)
y &lt;- c(&quot;a&quot;,&quot;b&quot;) #character vector
z &lt;- c(x,y) #new vector will contain elements of both vectors
z
[1] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;a&quot; &quot;b&quot;
class(z)
[1] &quot;character&quot; #numeric values will be converted into character string. As a vector contains same data type
</code></pre>

<p><strong>Vector arithmetic's</strong></p>
<pre><code class="R">#each member of vector will be multiplied by 5
    a &lt;- c(1,3,5)
    5*a
    [1] 5 15 25
#Addition or multiplication will be performed element by element
    b &lt;- c(2,4,6)
    a + b
    [1] 3 7 11
    a * b
    [1] 2 12 30
#multiplication as &quot;power to&quot; will be performed element by element
    a^2
    [1]1 9 25  # power 2 for each member
#if different length of vectors: then shorter will recycle - Recycling Rule.
    x &lt;- c(2,4)
    y &lt;- c(3,5,7,9)
    x + y
    [1] 5 9
</code></pre>

<p><strong>Accessing Vector data</strong>
<em> The values of a vector can be retrieved <code>&lt;variable&gt;[&lt;position&gt;]</code> or removed <code>&lt;variable&gt;[-&lt;position&gt;]</code>
</em> Use <code>c()</code> function to retrieve more then one value
* Use <code>&lt;-</code> to redirect the values in new variables</p>
<pre><code class="R">    x &lt;- c(&quot;a&quot;,&quot;b&quot;,&quot;c&quot;,&quot;d&quot;)
    x[3] #retrive command
    [1] c
    x[-3] # remove command
    [1] &quot;a&quot; &quot;b&quot; &quot;d&quot;
# retrieving more values frome one vector
    x[c(2,3)] #use the combine argument
    [1] &quot;b&quot; &quot;c&quot;
    x[c(2,3,3)]
    [1] &quot;b&quot; &quot;c&quot; &quot;c&quot; #Duplicate indexes
    x[c(2,1,3)]
    [1] &quot;b&quot; &quot;a&quot; &quot;c&quot; #out of order indexes
    x[2:4]
    [1] &quot;b&quot; &quot;c&quot; &quot;d&quot; #Range index with colon &quot;:&quot; operator
</code></pre>

<p><strong>Name Vector Members</strong>
* Names can be assigned to vector members by using <code>names( )</code></p>
<pre><code class="bash">    x &lt;- c(&quot;Christian&quot;, &quot;Brandt&quot;)
    names(x)&lt;- c(&quot;first_name&quot;, &quot;last_name&quot;) #naming each value entry of vector x
    x
     first_name    last_name
    &quot;Christian&quot;     &quot;Brandt&quot;
# data retrival works also with names
    x[c(&quot;last_name&quot;,&quot;first_name&quot;)]
    last_name     first_name
    &quot;Christian&quot;     &quot;Brandt&quot;
</code></pre>

<p><strong>Other commands</strong>
* such commands can all be combined like excel functions</p>
<pre><code class="R">#Create a sequential vector from 10 to 20 steps by 2
    seq(from=10,to=20,by=2)
    [1] 10 12 14 16 18 20
#Sequential vector of vector_length 5 which starts from 1 and incremented by 3
    seq(from=1,by=3,length.out=5)
    [1] 1 4 7 10 13
#replication command
    rep(&quot;hello&quot;,5)
    [1] &quot;hello&quot; &quot;hello&quot; &quot;hello&quot; &quot;hello&quot; &quot;hello&quot;
#combine rep( ) functions
    c(rep(1,2),rep(2,3),rep(3,4))
    [1] 1 1 2 2 2 3 3 3 3
#advanced examples of vector h    
    x&lt;- 1:10
    x
    [1] 1 2 3 4 5 6 7 8 9 10
    y &lt;- x[x&lt;8 &amp; x&gt;3]
    y
    [1] 4 5 6 7
</code></pre>

<hr />
<h3 id="matrices">Matrices</h3>
<ul>
<li>Creating a Matrix using <code>matrix</code> e. g.:
 <code>matrix(c(&lt;values&gt;),nrow= ,ncol=, byrow=TRUE/FALSE)</code></li>
<li>leave <code>nrow</code> or <code>ncol</code> blank e.g. only <code>ncol=2, byrow=TRUE</code> to fill the table by creating 2 columns and as many rows as needed for the data</li>
<li>combining matrices with the <code>cbind()</code> e. g. <code>cbind(x,y)</code></li>
</ul>
<pre><code class="R">#Create a matrix of 2 rows and 3 columns; byrow=FALSE fills columns first
    z &lt;- matrix ( c(1,2,3,4,5,6), nrow=2, ncol=3, byrow= TRUE)
    z
    [,1] [,2] [,3]
    [1,]  1    2    3
    [2,]  4    5    6
#Accessing the data
    x[2,3]
    [1] 6 # value of 2nd row and 3th column
    x[2,]
    [1] 2 4 6 # values of 2nd row
    x[,c(1,3)]
        [,1] [,3]
    [1,]  1    5
    [2,]  2    6
</code></pre>

<p><strong>Naming a matrix</strong>
<em> using <code>dimnames</code>
</em> then also retrivable by name e.g. <code>x["row1", c("col1","col2")]</code></p>
<pre><code class="R">dimnames(x)&lt;- list( c(&quot;row1&quot;, &quot;row2&quot;), c(&quot;col1&quot;, &quot;col2&quot;, &quot;col3&quot;) )
x
     col1 col2 col3
row1    1    2    3
row2    4    5    6
</code></pre>

<p><strong>Show properties of matrix</strong></p>
<pre><code class="R">class(x)
typeof(x)
nrow(x)
ncol(x)
length(x)
</code></pre>

<hr />
<h3 id="dataframe-this-is-the-excel-of-r">Dataframe - This is the Excel of R</h3>
<ul>
<li><strong>you want to use this</strong></li>
<li>each column is a class like in SPSS or other databases)</li>
<li>create a dataframe using <code>data.frame</code></li>
<li>each column is a vector datafile</li>
</ul>
<pre><code class="R">a&lt;- c(1,2,3) # numeric vector
b&lt;- c(&quot;x&quot;,&quot;y&quot;,&quot;z&quot;) # character vector
c&lt;- c(TRUE,FALSE,TRUE) # logical vector
# creating data frame
df&lt;- data.frame (a,b,c)
# accessing data frame
df
      a b     c # header contains vector names
    1 1 x TRUE
    2 2 y FALSE
    3 3 z TRUE
#extracting data
df[&quot;1&quot;,&quot;b&quot;]
[1] x
</code></pre>

<p><strong>Adding and removing data</strong>
* using <code>rbind</code> for rows and <code>cbind</code> for columns</p>
<pre><code class="R">#creating a dataframe    
    df &lt;- data.frame(first = c('Christian'), last = c('Brandt'), lucky_number = c(5), stringsAsFactors = FALSE)
    df
        first   last lucky_number
    Christian Brandt            5
#adding a row
    df &lt;- rbind(df, list('Erik', 'Bongcam', 10 ))
    df
            first    last lucky_number
    1   Christian  Brandt            5
    2        Erik Bongcam           10
#adding a column
    df &lt;-cbind(df, coffeetime =c(TRUE,TRUE))
# removal of data with
    df1&lt;- df[-2,] # remove 2nd row
    df2&lt;- df[,-3] # remove 3rd column
</code></pre>

<p><strong>Assigning names</strong></p>
<pre><code class="R"># You can assign names to rows and columns as:
    dimnames(df)&lt;- list ( c(&quot;row1&quot;,&quot;row2&quot;,&quot;row3&quot;), c(&quot;col1&quot;,&quot;col2&quot;,&quot;col3&quot;) )
    df
    col1 col2  col3
    row1    1  TRUE
    row2    2 FALSE
    row3    3  TRUE
# possible ways to get data (all with the same results)
    df[[3]]
    df[[&quot;col3&quot;]]
    df$col3
    df[,&quot;col3&quot;]

    [1] TRUE FALSE TRUE #result for all 4 commands
#retrieve multiple columns
    df[c(&quot;col2&quot;,&quot;col3&quot;)]
</code></pre>

<hr />
<h3 id="list">List</h3>
<ul>
<li>use the <code>list()</code> command to store objects in a data list</li>
<li>Each object (vector, matrix etc.) gets a own ID which is <code>[[1]]</code>, <code>[[2]]</code> and so on.</li>
</ul>
<pre><code class="R">    my_list &lt;- list(a,b,33) #a and b is a vector, 33 is only a value
    my_list
    [[1]]
    [1] 1 2 3

    [[2]]
    [1] &quot;SLU&quot; &quot;Uppsala&quot;

    [[3]]
    [1] 33
#Accessing data
    my_list[2]
    [[2]]
    [1] &quot;SLU&quot; &quot;Uppsala&quot;

    my_list[[2]][1]
    [1] &quot;SLU&quot;
#Modify the contents of list members directly with [[ ]]
    my_list[[2]][1] = &quot;new_SLU&quot;
</code></pre>

<h1 id="advanced-data-manipulation">Advanced data manipulation</h1>
<h2 id="if-else-for-while-function">if -else, for, while function</h2>
<p><strong>IF ELSE functions</strong>
* example:</p>
<pre><code class="R">x &lt;- 9
if (x &gt;= 10)
{
print(&quot;x is greater than or equal to 10&quot;)
}
else
{
print(&quot;x is less than 10&quot;)
}
</code></pre>

<p><strong>Loop functions</strong>
* <code>for(i in &lt;variable&gt;)</code>. Thats the counter. A interactive tutorial on loops can be found <a href="https://www.datacamp.com/community/tutorials/tutorial-on-loops-in-r#gs.7B0m9Qk">here</a> (includes coding, while functions).</p>
<pre><code class="R">for(i in 1:10) {print(i)} #output goes from 1 to 10
# or do stuff like this:
m&lt;- c(1:10) #a vector from 1 to 10
for(i in m){print(paste(&quot;5*&quot;,i,&quot;=&quot;,5*i))}
#results is 5*3=15 for one of the 10 rows
</code></pre>

<h2 id="working-dir">Working dir</h2>
<ul>
<li><code>getwd()</code> to get your current working dir</li>
<li><code>setwd("/Users/naat0001/Desktop/")</code> to specify were your workdir should be now</li>
<li>You can do this also in rStudio using the GUI</li>
</ul>
<p>Advanced folder creation with IF loop:</p>
<pre><code class="R">mainDir &lt;- &quot;/Users/naat0001/Desktop&quot;
subDir &lt;- &quot;outputDirectory&quot;
if (file.exists(subDir)) #TRUE or FALSE
{
setwd(file.path(mainDir, subDir))
}
else
{
dir.create(file.path(mainDir, subDir)) #creates dir
setwd(file.path(mainDir, subDir))
}
# file.path(mainDir, subDir) says basically
# /Users/naat0001/Desktop/outputDirectory   
</code></pre>

<h2 id="file-import-and-download">File import and download</h2>
<ul>
<li>Use delimiter files with <code>tab</code> <code>;</code> <code></code> <code>,</code> like <code>*.csv</code></li>
<li>Use <code>read.table</code> with sep="" (enter seperator here) and <code>header=TRUE</code></li>
</ul>
<pre><code class="R">x &lt;- read.table(&quot;path_to_file/file_name.csv&quot;, sep=&quot;,&quot;, header = TRUE)
# read.table needs sep=&quot;&quot; and header =TRUE
</code></pre>

<p><strong>Download and read a file</strong>
<code>download.file</code> does store the file in your working dir with <code>destfile=</code></p>
<pre><code class="R">download.file(&quot;https://raw.githubusercontent.com/swcarpentry/r-novice-gapminder/gh-pages/_episodes_rmd/data/gapminder-FiveYearData.csv&quot;, destfile = &quot;gapminder-FiveYearData.csv&quot;)
# read the file to a varible
x &lt;- read.csv(&quot;gapminder-FiveYearData.csv&quot;)
# you can also directly read the online file to a variable
x &lt;- read.table(&quot;https://raw.githubusercontent.com/swcarpentry/r-novice-gapminder/gh-pages/_episodes_rmd/data/gapminder-FiveYearData.csv&quot;, sep=&quot;,&quot;, header = TRUE)
</code></pre>

<p><strong>Useful commands</strong></p>
<pre><code class="R">#x is a dataframe
str(x) # tells you what x is, e.g. a dataframe and what class in each column is stored (e.g. numerical)
head(x) #To view the first few rows of the data
tail(x) #To view the last few rows of the data
summary(x) #shows for each coloum various stuff like minimum, lower quartile, median, mean, upper quartile...
nrow(x) #number of rows
#nrow example with build in functions:
nrow(x[x$BPressure==&quot;high&quot; | x$BPressure==&quot;pre-high&quot;,]) #counts rows in colum BPressure with the value high oder pre-high
#like SVERWEIS if value in BCHolesterol &gt;190 show value of column Name in the same row  
x[x$BCholesterol&gt;190,&quot;Name&quot;]
</code></pre>

<p><strong>Export data frames into a file</strong>
<em> You do this with the <code>write.table()</code> function
</em> If you don’t have any row names set up for your data frame use this:</p>
<pre><code class="R">write.table(data,file=&quot;out.txt&quot;,sep=&quot;\t&quot;,quote=FALSE,row.names=FALSE)
</code></pre>

<ul>
<li>If you do have row names then use the command below which will keep the row names and will move the column names so they line up correctly with the data.</li>
</ul>
<pre><code class="R">write.table(data,file=&quot;out.txt&quot;,sep=&quot;\t&quot;,quote=FALSE,col.names=NA)`
</code></pre>

<h2 id="merging-two-files">Merging two files</h2>
<ul>
<li>using the <code>merge</code> command</li>
<li>you tell them where the ID is (like Excel SVERWEIS)</li>
</ul>
<pre><code class="R"># lets say i have 2 dataframes:
df1 &lt;- data.frame(Chr=paste0('chr',1:9), Gene=paste0('gene',19:11))
df2 &lt;- data.frame(Chr=paste0('chr',2:10),Position=paste0('pos',22:30))
# they look like this, shortend it with (...)
&gt; df1
   Chr   Gene
1 chr1 gene19
2 chr2 gene18
3 chr3 gene17
4 chr4 gene16
(...)
&gt; df2
    Chr Position
1  chr2    pos22
2  chr3    pos23
3  chr4    pos24
4  chr5    pos25
(...)
# you need two uniq column names, one in each dataframe
# we merge them based on the uniq &quot;Chr&quot; column (our &quot;ID&quot;)
df3 &lt;- merge(df1, df2, by=&quot;Chr&quot;)
# it only works if the column names are the same
# so use this to name them (changes the name of df2 column 1 to &quot;Chr_ID&quot;:
colnames(df2)[1] &lt;- &quot;Chr_ID&quot;
&gt; df3
 Chr   Gene Position
1 chr2 gene18    pos22
2 chr3 gene17    pos23
3 chr4 gene16    pos24
4 chr5 gene15    pos25
(...)

# alternatively say which columns should be used for the merging:
df3 &lt;- merge(df1, df2, by.x=&quot;Chr&quot;, by.y=&quot;Chr_ID&quot;)
</code></pre>

<h1 id="graphs-in-r">Graphs in R</h1>
<ul>
<li>Overview of some basic plots can be found <a href="https://www.statmethods.net/graphs/index.html">here</a>.</li>
<li>Advanced graphs can be found <a href="https://www.statmethods.net/advgraphs/index.html">here</a>.</li>
<li>This example is based on the hist plot:</li>
</ul>
<pre><code class="R">x &lt;- rnorm(50) # 50 random values from normal distribution.
hist(x) # plot histogram of values of x
hist(x,main=&quot;title of histogram&quot;) # change title
hist(x, main = &quot;Title of histogram&quot;, col = &quot;orange&quot;) # with color bars

# How to save / export graphs from console in pdf format.
pdf(file=&quot;my_histogram.pdf&quot;)
hist(x, main = &quot;Title of histogram&quot;, col = &quot;orange&quot;)
dev.off()
</code></pre>

<ul>
<li><a href="https://cran.r-project.org/doc/manuals/r-release/R-intro.html">An introduction to R</a></li>
<li><a href="http://r4ds.had.co.nz">R for data science</a></li>
</ul></div>
        </div>

        <footer class="col-md-12">
            <hr>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script>
            var base_url = "../..",
                shortcuts = {"search": 83, "help": 191, "next": 78, "previous": 80};
        </script>
        <script src="../../js/base.js" defer></script>
        <script src="../../search/main.js" defer></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                <h4 class="modal-title" id="exampleModalLabel">Search</h4>
            </div>
            <div class="modal-body">
                <p>
                    From here you can search these documents. Enter
                    your search terms below.
                </p>
                <form role="form">
                    <div class="form-group">
                        <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query" title="Type search term here">
                    </div>
                </form>
                <div id="mkdocs-search-results"></div>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div><div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="Keyboard Shortcuts Modal" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                <h4 class="modal-title" id="exampleModalLabel">Keyboard Shortcuts</h4>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>
