<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="../../img/favicon.ico">
        <title>Metagenomics - Bioinformatics</title>
        <link href="../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../css/font-awesome.min.css" rel="stylesheet">
        <link href="../../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/color-brewer.min.css">
        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
        <![endif]-->

        <script src="../../js/jquery-1.10.2.min.js" defer></script>
        <script src="../../js/bootstrap-3.0.3.min.js" defer></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
        <script>hljs.initHighlightingOnLoad();</script> 
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
            <div class="container">

                <!-- Collapsed navigation -->
                <div class="navbar-header">
                    <!-- Expander button -->
                    <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                        <span class="sr-only">Toggle navigation</span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                        <span class="icon-bar"></span>
                    </button>
                    <a class="navbar-brand" href="../..">Bioinformatics</a>
                </div>

                <!-- Expanded navigation -->
                <div class="navbar-collapse collapse">
                        <!-- Main navigation -->
                        <ul class="nav navbar-nav">
                            <li class="dropdown">
                                <a href="#" class="dropdown-toggle" data-toggle="dropdown">Shell <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
  <li class="dropdown-submenu">
    <a href="#">Basics</a>
    <ul class="dropdown-menu">
            
<li >
    <a href="../../shell/shell/">Commands</a>
</li>
            
<li >
    <a href="../../shell/shell_adv/">Cloud Computing</a>
</li>
    </ul>
  </li>
                                    
<li >
    <a href="../../shell/fasta/">File Manipulation (fasta)</a>
</li>
                                    
<li >
    <a href="../../shell/git/">Git</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="dropdown-toggle" data-toggle="dropdown">Python <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li >
    <a href="../../python/python/">Basics</a>
</li>
                                    
<li >
    <a href="../../python/bioconda/">Bioconda</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="dropdown-toggle" data-toggle="dropdown">R <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li >
    <a href="../../R/R/">Basics</a>
</li>
                                    
  <li class="dropdown-submenu">
    <a href="#">Graphics</a>
    <ul class="dropdown-menu">
            
<li >
    <a href="../../R/ggplot/">ggplot</a>
</li>
    </ul>
  </li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="dropdown-toggle" data-toggle="dropdown">Tools <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li >
    <a href="../../tools/atom/">Atom Editor</a>
</li>
                                    
<li >
    <a href="../../tools/emboss/">EMBOSS</a>
</li>
                                    
<li >
    <a href="../../tools/mkdocs/">mkdocs</a>
</li>
                                    
<li >
    <a href="../../tools/UGENE/">UGENE</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="dropdown-toggle" data-toggle="dropdown">Sequencing <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li >
    <a href="../../sequencing/programs/">Programs</a>
</li>
                                    
<li >
    <a href="../../sequencing/illumina/">HTS (Illumina)</a>
</li>
                                    
<li >
    <a href="../../sequencing/nanopore/">Longreads (Nanopore)</a>
</li>
                                    
<li >
    <a href="../../sequencing/hybrid/">Hybridassembly</a>
</li>
                                    
<li >
    <a href="../../sequencing/annotation/">Annotation</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown active">
                                <a href="#" class="dropdown-toggle" data-toggle="dropdown">Metagenome <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li class="active">
    <a href="./">Metagenomics</a>
</li>
                                    
<li >
    <a href="../metabarcoding/">Metabarcoding</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="dropdown-toggle" data-toggle="dropdown">Docker <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li >
    <a href="../../docker/docker/">Docker-Basics</a>
</li>
                                </ul>
                            </li>
                            <li class="dropdown">
                                <a href="#" class="dropdown-toggle" data-toggle="dropdown">Other Topics <b class="caret"></b></a>
                                <ul class="dropdown-menu">
                                    
<li >
    <a href="../../other_topics/databases/">Databases</a>
</li>
                                    
<li >
    <a href="../../other_topics/gwas/">GWAS</a>
</li>
                                    
<li >
    <a href="../../other_topics/gs/">GS</a>
</li>
                                    
<li >
    <a href="../../other_topics/RNA_seq/">RNA seq</a>
</li>
                                </ul>
                            </li>
                        </ul>

                    <ul class="nav navbar-nav navbar-right">
                        <li>
                            <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                                <i class="fa fa-search"></i> Search
                            </a>
                        </li>
                            <li >
                                <a rel="next" href="../../sequencing/annotation/">
                                    <i class="fa fa-arrow-left"></i> Previous
                                </a>
                            </li>
                            <li >
                                <a rel="prev" href="../metabarcoding/">
                                    Next <i class="fa fa-arrow-right"></i>
                                </a>
                            </li>
                    </ul>
                </div>
            </div>
        </div>

        <div class="container">
                <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
        <li class="main active"><a href="#metagenomics">Metagenomics</a></li>
            <li><a href="#overview-terminology">Overview &amp; Terminology</a></li>
            <li><a href="#whole-metagenome-sequencing-practical">Whole Metagenome Sequencing - practical</a></li>
            <li><a href="#metagenome-assembly-and-binning-practical">Metagenome assembly and binning - practical</a></li>
    </ul>
</div></div>
                <div class="col-md-9" role="main">

<h1 id="metagenomics">Metagenomics</h1>
<h2 id="overview-terminology">Overview &amp; Terminology</h2>
<ul>
<li>collected genomic information of a sample</li>
<li>unbiased targeting (no special marker like metabarcoding)</li>
<li>Metabarcoding (16S) is used for large number of samples, i.e., multiple patients, longitudinal studies, etc. but offers limited taxonomical and functional resolution in comparision</li>
<li><strong>Whole Metagenome sequencing (WMS)</strong>, or shotgun metagenome sequencing, provides insight into community biodiversity and function in contrast to Metabarcoding, where only a specific region of the bacterial community (the 16s rRNA) is sequenced</li>
<li><strong>WMS</strong> aims at sequencing all the genomic material present in the environment</li>
<li><strong>WMS</strong> is more expensive but offers increased resolution, and <strong>allows the discovery of viruses as well as other mobile genetic elements</strong></li>
</ul>
<p><strong>Microbiota</strong>
<em> organism within a set environment
</em> mainly focused on bacteria
* exluding viruses &amp; mobile elements</p>
<p><strong>Microbiome</strong>
<em> includes all factors of an environment
</em> (organism, genetic elements, metabolites, proteins)</p>
<p><strong>Holobiont</strong>
<em> assemblages of different species that form a ecological unit
</em> divided into: virome, microbiome and macrobiological hosts</p>
<h2 id="whole-metagenome-sequencing-practical">Whole Metagenome Sequencing - practical</h2>
<ul>
<li>comparing samples from the Pig Gut Microbiome against the Human Gut Microbiome</li>
</ul>
<p>Software used:
-   <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>
-   <a href="https://ccb.jhu.edu/software/kraken/">Kraken</a>
-   <a href="https://www.r-project.org/">R</a>
-   <a href="https://github.com/fbreitwieser/pavian">Pavian</a></p>
<p>Data used:</p>
<pre><code class="bash">curl -O -J -L https://osf.io/h9x6e/download
tar xvf subset_wms.tar.gz
</code></pre>

<ul>
<li>we <code>ssh</code> using additionally the <code>-X</code> flag to allow a graphical interface</li>
<li>we used <code>fastqc</code> do check the quality of our data</li>
</ul>
<h3 id="1-taxonomic-read-assignment-with-kraken">1. Taxonomic read assignment with <code>Kraken</code></h3>
<ul>
<li>first we need the Kraken database</li>
</ul>
<pre><code class="bash">#create a databases directory and download the database
wget https://ccb.jhu.edu/software/kraken/dl/minikraken_20171019_4GB.tgz
tar xzf minikraken_20171019_4GB.tgz
</code></pre>

<ul>
<li>running <code>kraken</code> on the reads</li>
<li><strong>produces a tab-delimited file with an assigned TaxID for each read</strong></li>
</ul>
<pre><code class="bash"># for the script you need to declare the KRAKEN_DB path
KRAKEN_DB=&quot;/mnt/databases/minikraken_20171013_4GB&quot;
for i in *_1.fastq
do
    prefix=$(basename $i _1.fastq)
    # print which sample is being processed
    echo $prefix
    kraken --db $KRAKEN_DB --threads 2 --fastq-input ${prefix}_1_corr.fastq ${prefix}_2_corr.fastq &gt; /home/student/wms/results/${prefix}.tab
    kraken-report --db $KRAKEN_DB /home/student/wms/results/${prefix}.tab &gt; /home/student/wms/results/${prefix}_tax.txt
done
</code></pre>

<h3 id="2-visualization-with-pavian-in-r">2. Visualization with Pavian in R</h3>
<ul>
<li>Installing and run Pavian in R with those commands:</li>
</ul>
<pre><code class="R">#one time only installations and setup
options(repos = c(CRAN = &quot;http://cran.rstudio.com&quot;))
if (!require(remotes)) { install.packages(&quot;remotes&quot;) }
remotes::install_github(&quot;fbreitwieser/pavian&quot;)
#Start up command
pavian::runApp(port=5000)
</code></pre>

<ul>
<li>creates a new R-tab were you can open the kraken files</li>
<li>creates a nice sankey chart of your organism within your sample</li>
<li>its scalable, allows comparision between samples (as a table) and indepth analysis of samples (see figure)</li>
</ul>
<blockquote>
<p>Example results in one sample as a sankey chart using pavian
<img alt="" src="https://i.imgur.com/Lmvswjo.png" /></p>
</blockquote>
<ul>
<li>KRONA chart is also for visualization of metagenomic data</li>
<li>good for explanation of your sample</li>
<li>but PAVIAN is way better</li>
</ul>
<h2 id="metagenome-assembly-and-binning-practical">Metagenome assembly and binning - practical</h2>
<p>AIM: inspect and assemble metagenomic data and retrieve draft genomes</p>
<ul>
<li>using a mock community of 20 bacteria simulating Illumina HiSeq created with <a href="http://insilicoseq.readthedocs.io">InSilicoSeq</a></li>
<li>selected from the <a href="http://ocean-microbiome.embl.de/companion.html">Tara Ocean study</a> that recovered 957 distinct Metagenome-assembled-genomes (or MAGs) that were previsouly unknown</li>
</ul>
<h3 id="1-correction-and-assembly-of-the-illumina-reads">1. Correction and assembly of the illumina reads</h3>
<ul>
<li>check the fastq data with <code>fastqc</code></li>
<li>do <code>sickle</code> and <code>scythe</code> as needed; see <a href="../../sequencing/illumina/">HTS</a></li>
<li>assembly here with <code>megahit</code></li>
</ul>
<pre><code class="bash">sickle pe -f tara_reads_R1.fastq.gz -r tara_reads_R2.fastq.gz -t sanger -o tara_trimmed_R1.fastq -p tara_trimmed_R2.fastq -s /dev/null
megahit -1 tara_trimmed_R1.fastq -2 tara_trimmed_R2.fastq -o tara_assembly
</code></pre>

<h3 id="2-binning">2. Binning</h3>
<ul>
<li>First map the reads back against the assembly (for coverage information)</li>
<li>bowtie2 is a ultra fast short read mapper/aligner</li>
</ul>
<pre><code class="bash">ln -s tara_assembly/final.contigs.fa . #link contigs to current folder
bowtie2-build final.contigs.fa final.contigs #index
bowtie2 -x final.contigs -1 tara_trimmed_R1.fastq -2 tara_trimmed_R2.fastq | samtools view -bS -o tara_to_sort.bam #alignment and bam conversion
samtools sort tara_to_sort.bam -o tara.bam
samtools index tara.bam
</code></pre>

<ul>
<li>then we run <code>metabat</code></li>
</ul>
<pre><code class="bash">runMetaBat.sh -m 1500 final.contigs.fa tara.bam
mv final.contigs.fa.metabat-bins1500 metabat
</code></pre>

<h3 id="3-qc-of-the-bins">3. QC of the bins</h3>
<ul>
<li>first time you run <code>checkm</code> you have to create the database</li>
</ul>
<pre><code class="bash">sudo checkm data setRoot ~/.local/data/checkm
</code></pre>

<pre><code class="bash">checkm lineage_wf -x fa metabat checkm/
checkm bin_qa_plot -x fa checkm metabat plots
</code></pre>

<h3 id="4-get-organism-information-and-plot-it">4. Get organism information and plot it</h3>
<pre><code class="bash">#tells you which organism, saves under checkm/lineage needs folder checkm
checkm qa checkm/lineage.ms checkm
#plots its
checkm bin_qa_plot -x fa checkm metabat plots
</code></pre>

<ul>
<li>you get something like this, which should represent 10 different species (theoretically):
<img alt="" src="https://i.imgur.com/RG2mmkR.png" /></li>
<li>now you can annotate each bin (fasta files are in metabat dir)</li>
<li>more informations can be found <a href="https://www.nature.com/articles/s41564-017-0012-7">here</a> and <a href="https://www.nature.com/articles/sdata2017203">here</a></li>
</ul>
<h3 id="5-barrnap">5. Barrnap</h3>
<ul>
<li><strong>BA</strong>sic <strong>R</strong>apid <strong>R</strong>ibosomal <strong>RNA</strong> <strong>P</strong>redictor</li>
<li>isnt tsuitable for novel stuff since it has to be in the database, but seems still to work (see figure above of unknown marine bacteria)</li>
<li>you can download it <a href="https://github.com/tseemann/barrnap">here</a></li>
</ul></div>
        </div>

        <footer class="col-md-12">
            <hr>
            <p>Documentation built with <a href="https://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>
        <script>
            var base_url = "../..",
                shortcuts = {"help": 191, "next": 78, "previous": 80, "search": 83};
        </script>
        <script src="../../js/base.js" defer></script>
        <script src="../../search/main.js" defer></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                <h4 class="modal-title" id="exampleModalLabel">Search</h4>
            </div>
            <div class="modal-body">
                <p>
                    From here you can search these documents. Enter
                    your search terms below.
                </p>
                <form role="form">
                    <div class="form-group">
                        <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query" title="Type search term here">
                    </div>
                </form>
                <div id="mkdocs-search-results"></div>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div><div class="modal" id="mkdocs_keyboard_modal" tabindex="-1" role="dialog" aria-labelledby="Keyboard Shortcuts Modal" aria-hidden="true">
    <div class="modal-dialog">
        <div class="modal-content">
            <div class="modal-header">
                <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                <h4 class="modal-title" id="exampleModalLabel">Keyboard Shortcuts</h4>
            </div>
            <div class="modal-body">
              <table class="table">
                <thead>
                  <tr>
                    <th style="width: 20%;">Keys</th>
                    <th>Action</th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td class="help shortcut"><kbd>?</kbd></td>
                    <td>Open this help</td>
                  </tr>
                  <tr>
                    <td class="next shortcut"><kbd>n</kbd></td>
                    <td>Next page</td>
                  </tr>
                  <tr>
                    <td class="prev shortcut"><kbd>p</kbd></td>
                    <td>Previous page</td>
                  </tr>
                  <tr>
                    <td class="search shortcut"><kbd>s</kbd></td>
                    <td>Search</td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div class="modal-footer">
            </div>
        </div>
    </div>
</div>

    </body>
</html>
